### ISSUE TITLE
Testing Output Network

### ISSUE THREAD

**Issue by AleLdf** [dahyun-kang/renet#4]
Hi,
I'd like to give an image input from testset, in order to visualize the result (the corrisponding label).  
How can I do this? Is there a command?
Thanks

**Comment by dahyun-kang** [comment]
Hello,

This code base does not officially provide such function at this point.
Unfortunately, it seems unlikely to be easily implemented from this repository. 🤔
You may need to load and transform your own query image, and replace it with the query set provided in the dataloader.

For example, the code snippet from `test.py`
https://github.com/dahyun-kang/renet/blob/b58ebc092fcdb40e7f534f6407512df4f109cacd/test.py#L28-L35

should be something like below:
```python
        for i, (data, labels) in enumerate(tqdm_gen, 1):
            myquery = # read, transform, and load the image to the GPU here.
            data = data.cuda()
            data = torch.cat((data[:k], myquery), dim=0)
            data = model(data)
            data_shot, data_query = data[:k], data[k:]
            model.module.mode = 'cca'
            logits = model((data_shot.unsqueeze(0).repeat(args.num_gpu, 1, 1, 1, 1), data_query))
            print(logits.argmax(dim=1))  # you can see the prediction out of 0 to N-1 here
```

In this case, you can see how your query image is predicted based on a various compositions of support set.
Please let me know if I understood your question incorrectly.

Have a great day! :smiley:

**Comment by AleLdf** [comment]
I had tried to give the following input : myquery= 'name.jpg' taken from testset. Then I had tried to give an other input: 
 myquery=' path to name.jpg'. In both cases I had the same following error:
 _TypeError: expected Tensor as element 1 in argument 0, but got str_ 
It seems that it doesn't recognize the input myquery.
Thanks

**Comment by lyxdlut** [comment]
have you solved the question? can you tell me how to visualize the result?
